src/components/views/settings/SetIdServer.js

```diff
@@ -0,0 +1,194 @@
+
+    _onIdentityServerChanged = (ev) => {
+        const u = ev.target.value;
+
+        this.setState({idServer: u});
+    };
+
+    _getTooltip = () => {
+        if (this.state.busy) {
+            const InlineSpinner = sdk.getComponent('views.elements.InlineSpinner');
+            return <div>
+                <InlineSpinner />
+                { _t("Checking Server") }
+            </div>;
+        } else if (this.state.error) {
+            return this.state.error;
+        } else {
+            return null;
+        }
+    };
+
+    _idServerChangeEnabled = () => {
+        return !!this.state.idServer && !this.state.busy;
+    };
+
+    _saveIdServer = async () => {
+        this.setState({busy: true});
+
+        const fullUrl = unabbreviateUrl(this.state.idServer);
+
+        const errStr = await checkIsUrl(fullUrl);
+
+        let newFormValue = this.state.idServer;
+        if (!errStr) {
+            MatrixClientPeg.get().setIdentityServerUrl(fullUrl);
+            localStorage.removeItem("mx_is_access_token");
+            localStorage.setItem("mx_is_url", fullUrl);
+            newFormValue = '';
+        }
+        this.setState({
+            busy: false,
+            error: errStr,
+            currentClientIdServer: MatrixClientPeg.get().getIdentityServerUrl(),
+            idServer: newFormValue,
+        });
+    };
+
+    render() {
+        const idServerUrl = this.state.currentClientIdServer;
+        let sectionTitle;
+        let bodyText;
+        if (idServerUrl) {
+            sectionTitle = _t("Identity Server (%(server)s)", { server: abbreviateUrl(idServerUrl) });
+            bodyText = _t(
+                "You are currently using <server></server> to discover and be discoverable by " +
+                "existing contacts you know. You can change your identity server below.",
+                {},
+                { server: sub => <b>{abbreviateUrl(idServerUrl)}</b> },
+            );
+        } else {
+            sectionTitle = _t("Identity Server");
+            bodyText = _t(
+                "You are not currently using an Identity Server. " +
+                "To discover and be discoverable by existing contacts you know, " +
+                "add one below",
+            );
+        }
+
+        return (
+            <form className="mx_SettingsTab_section mx_SetIdServer" onSubmit={this._saveIdServer}>
+                <span className="mx_SettingsTab_subheading">
+                    {sectionTitle}
+                </span>
+                <span className="mx_SettingsTab_subsectionText">
+                    {bodyText}
+                </span>
+                <Field label={_t("Identity Server")}
+                    id="mx_SetIdServer_idServer"
+                    type="text" value={this.state.idServer} autoComplete="off"
+                    onChange={this._onIdentityServerChanged}
+                    tooltip={this._getTooltip()}
+                />
+                <input className="mx_Dialog_primary"
```

Comment:
This should be an AccessibleButton with `kind="primary_sm"` to match the design.

Reply:
ok, but this means we can't use the native form element and have to faff about using an AccessibleButton and then putting back the submit-on-enter behaviour. If we use native elements, the browser does it for us for free.

Reply:
No? A `type=submit` here and a preventDefault /stopPropagation in the handler and you're done.

Reply:
OK, but this seem like a really roundabout way of using HTML (ie. make a `div` then make it behave like a submit button rather than putting the appropriate class on a submit button)

Reply:
The component does support changing its type too. There's nothing wrong with sticking `elem='input'` or `elem='button'` on it.